### PR TITLE
Add test cases for Note

### LINKED ISSUE #131: Task: Add test cases for Note
We should test the Note class, specifically the creation of `Note` with null value, and its `equals` method.

### PR DESCRIPTION
**Pull request by ruijietay**
Referencing #131, added some test cases for `Note` class.

Resolves #131

### CODE REVIEW

**Review by RyanNgWH — CHANGES_REQUESTED** [review]
Present test cases looks pretty good! However, should we test the `isMatch()` method as well?

File: src/test/java/seedu/address/model/person/NoteTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Test
+    public void isMatch() {
+        Note note = new Note("Likes to eat");
+
+        // Exact match -> returns true
+        assertTrue(note.isMatch("Likes to eat"));
+
+        // Substring whole word -> returns true
+        assertTrue(note.isMatch(" Likes"));
```
Comment (RyanNgWH) [inline src/test/java/seedu/address/model/person/NoteTest.java:45]: Should we remove the whitespace in front of `" Likes"`? Each test case should test only 1 thing, whitespaces is being tested in another test case below.

Comment (RyanNgWH) [reply src/test/java/seedu/address/model/person/NoteTest.java:45]: Same with the next 2 test cases

**Review by RyanNgWH — CHANGES_REQUESTED** [review]
Left a suggestion about the test cases. The overall logic looks good!

File: src/test/java/seedu/address/model/person/NoteTest.java (hunk 1 above)
Comment (ruijietay) [reply src/test/java/seedu/address/model/person/NoteTest.java]: Great catch 🤠 Fixed

File: src/main/java/seedu/address/model/person/Note.java [hunk 2]
```diff
@@ -36,7 +36,7 @@ public boolean isMatch(Object otherValue) {
 
         String other = (String) otherValue;
 
-        return this.getValue().toLowerCase().contains(other.toLowerCase());
+        return this.getValue().trim().toLowerCase().contains(other.trim().toLowerCase());
```
Comment (jjchee77) [inline src/main/java/seedu/address/model/person/Note.java:39]: The `parseNote` method in `ParserUtil.java` already contains the logic to trim the note before instantiating a `Note` instance. Seems redundant to add it here again.  However, `NoteCommandParser.java` does not appear to be using the `parseNote` method and instead creates a new note without trimming.

**Review by jjchee77 — APPROVED** [review]
Just one nit to pick, otherwise, good work on the comprehensive test cases, LGTM!

**Review by RyanNgWH — APPROVED** [review]
LGTM! Good catch of the test case issues in `AddressTest.java` as well! 👍🏼

**Pull request merged**